src/dyn/dyn_decoherence_methods.cpp

```diff
@@ -1521,7 +1633,7 @@ void update_forces_xf(dyn_variables& dyn_var, nHamiltonian& ham, nHamiltonian& h
   *dyn_var.f += *dyn_var.f_xf;
 }
 
-void propagate_half_xf(dyn_variables& dyn_var, nHamiltonian& Ham, dyn_control_params& prms){
+void propagate_half_xf(dyn_variables& dyn_var, nHamiltonian& Ham, dyn_control_params& prms, bp::dict params, int rotation){

```

Comment:
makes no sense to introduce "dict" parameter - on the C++ side, it is usually converted into dyn_control_param class before use; you can of course overload the "propagate_half_xf" function to take just the dict param - e.g. for convenience of Python users, but internally it would be just converting this argument to the "dyn_control_param". Also, consider having the "rotation" argument one of the user-specified control parameters - if that's the goal, tuck it under "dyn_control_param"

Reply:
Now, the params dictionary argument used for obtaining the current timestep is removed. The timestep variable (and time-dependent width) is managed as dynamical variables, so it computed separately from the electronic propagation.